src/components/Search/Search.js

```diff
@@ -392,7 +370,72 @@ const Search = () => {
             pane: 'imagery'
           }
         )
-          .addTo(clickedFootprintsImageLayer)
+          .addTo(clickedFootprintImageLayerRef.current)
+          .on('load', function () {
+            // hide loading spinner
+            dispatch(setSearchLoading(false))
+          })
+          .on('tileerror', function () {
+            console.log('Tile Error')
+          })
+      })
+  }
+
+  // function to remove old image layer and add new Tiler image layer to map
+  function addMosaic() {
+    // clear previous results from map
+    clearResults()
+
+    // show loading spinner
+    dispatch(setSearchLoading(true))
+
+    // const tilerParams = constructTilerParams(_collectionSelected)
+
+    // build date input
+    const datetime =
+      convertDateTimeForAPI(dateTimeRef.current[0]) +
+      '/' +
+      convertDateTimeForAPI(dateTimeRef.current[1])
+
+    // get viewport bounds and setup bbox parameter
+    const viewportBounds = map.getBounds()
+    const bbox = [
+      viewportBounds._southWest.lng,
+      viewportBounds._southWest.lat,
+      viewportBounds._northEast.lng,
+      viewportBounds._northEast.lat
+    ]
+
+    const swCorner = L.latLng(bbox[1], bbox[0])
+    const neCorner = L.latLng(bbox[3], bbox[2])
+    const mosaicBounds = L.latLngBounds(swCorner, neCorner)
+
+    const requestOptions = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/vnd.titiler.stac-api-query+json'
+      },
+      body: JSON.stringify({
+        stac_api_root: process.env.REACT_APP_STAC_API_URL,
+        asset_name: 'visual', // todo: use first entry in assets config
+        collections: [selectedCollectionRef.current],
+        datetime,
+        bbox,
+        max_items: 10
+        // todo: query w/ cloud_cover, as a JSON object rather than a url-encoded string
+      })
+    }
+    fetch(`${mosaicTilerURL}/mosaicjson/mosaics`, requestOptions)
+      .then((r) => r.json())
+      .then((body) => {
+        const tileHref = body?.links?.find((el) => el.rel === 'tiles').href
```

Comment:
```suggestion
        const tileHref = body?.links?.find((el) => el.rel === 'tiles')?.href
```

if it can't find the `tiles` ref, we don't want it to blow up with an attribute not found exception on the undefined value

Reply:
this should have a guard around it to only add the layer if if found the href

Reply:
This has been fixed and a check added.